test/Datadog.Trace.ClrProfiler.Managed.Tests/AttributeTests.cs

```diff
@@ -0,0 +1,55 @@
+// <copyright file="AttributeTests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using FluentAssertions;
+using Xunit;
+
+namespace Datadog.Trace.ClrProfiler.Managed.Tests
+{
+    public class AttributeTests
+    {
+        /// <summary>
+        /// This test reports instances in the Datadog.Trace.ClrProfiler.Managed and Datadog.Trace
+        /// where attributes use named arguments that are not a built-in type. Using a custom type
+        /// (especially one from the Datadog.Trace assembly) as a named argument can fail with a
+        /// System.Reflection.CustomAttributeFormatException if the owning assembly cannot be found
+        /// in the default load context and other assembly resolve events are registered, in addition
+        /// to the ones added by the Datadog .NET Tracer
+        /// </summary>
+        [Fact]
+        public void AttributesInstantiationsOnlyUseBuiltinTypes()
+        {
+            List<string> invalidAttributeUsages = new();
+            var coreAssembly = typeof(object).Assembly;
+
+            var tracerAssembly = typeof(Tracer).Assembly;
+            var managedAssembly = typeof(Instrumentation).Assembly;
+            var types = tracerAssembly.GetTypes().Concat(managedAssembly.GetTypes());
+
+            foreach (var type in types)
+            {
+                foreach (var member in type.GetMembers())
+                {
+                    foreach (var attribute in member.CustomAttributes)
+                    {
+                        foreach (var namedArg in attribute.NamedArguments)
```

Comment:
Just confirming, the issue is explicitly only around _named_ arguments right?  Ordered arguments don't cause any issues, even if they're custom types?

Reply:
I thought this issue happens on all kind of arguments with a custom Type regardless of named or not.

Reply:
Good point...it definitely fails on named arguments like the following declaration:

```csharp
[Duck(Name = "_binding", Kind = DuckKind.Field)]
```

To test constructor arguments I temporarily modified `DuckAttribute` to have a constructor like the following:
```csharp
        public DuckAttribute(string nameCtorArg, DuckKind kindCtorArg)
        {
            Name = nameCtorArg;
            Kind = kindCtorArg;
        }
```

To test positional arguments on the constructor, I used the following invocation but it succeeded.
```csharp
[Duck("_binding", DuckKind.Field)]
```

To test named arguments on the constructor, I used the following invocation and used a named argument of type `System.Reflection.BindingFlags` and it succeeds.
```csharp
[Duck("_binding", DuckKind.Field, BindingFlags = DuckAttribute.DefaultFlags | BindingFlags.IgnoreCase)]
```

And for one final test, I added a `DuckKind` named argument just to make sure, and sure enough it fails again with this new custom type.
```csharp
[Duck("_binding", DuckKind.Field, BindingFlags = DuckAttribute.DefaultFlags | BindingFlags.IgnoreCase, Kind = DuckKind.Field)]
```

Are there any other potential usages I'm missing?

Reply:
What was the conclusion of your tests? I would expect both named and ordered arguments to behave the same way, since that is handled by the C# compiler and it emits the same IL in both cases (no difference at runtime).

Reply:
My conclusion was that named arguments (i.e. object properties) are treated differently than constructor arguments. The error only occurs when a **named argument** has a custom type. Constructor arguments (both named and positional) do not cause errors when the argument type is a custom type.

Reply:
Interesting, thanks!

Reply:
Oops, I was thinking of named method arguments like `Foo(name: "bar")`, but these are attributes. Never mind!